src/dodal/common/beamlines/beamline_parameters.py

```diff
@@ -71,17 +72,45 @@ def parse_list_element(cls, value: str):
         else:
             return float(value)
 
+    @classmethod
+    def find_first_index(cls, value: str, chars: Iterable):
+        return min((value.find(c) for c in chars if c in value), default=-1)
+
+    @classmethod
+    def find_close_square_bracket(cls, value: str):
```

Comment:
What filetype are the beamline parameter files? Is it possible to convert to JSON or something that can be parsed by existing libraries?